lib/pbench/server/api/resources/query_apis/search_api.py

```diff
@@ -0,0 +1,224 @@
+from http import HTTPStatus
+from logging import Logger
+from typing import Any, AnyStr, Dict
+
+from flask import jsonify
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.query_apis import (
+    CONTEXT,
+    ElasticBase,
+    Parameter,
+    ParamType,
+    PostprocessError,
+    Schema,
+)
+
+
+class IndexSearchAPI(ElasticBase):
+    """
+    Create a search request based on given query parameter.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            Schema(
+                Parameter("user", ParamType.USER, required=False),
+                Parameter("start", ParamType.DATE, required=True),
+                Parameter("end", ParamType.DATE, required=True),
+                Parameter("search_term", ParamType.STRING, required=False),
+                Parameter("fields", ParamType.LIST, required=False),
+            ),
+        )
+
+    def flatten_json(self, es_json: Dict[AnyStr, Any]) -> Dict[AnyStr, Any]:
+        """
+        Helper function for Postprocess for flattening the json. Any exception raised here will be
+        delegated as a postprocess error.
+        We convert a nested data layer object into a new object with only one layer of key/value pairs.
+        Objects such as parent -> child are flattened as parent.child like following:
+        parent -> array -> [obj1,objN] are flattened as
+                parent.array.0 = obj1, parent.array.N = objN
```

Comment:
`[obj1, objN]` only has two elements, making the example a bit confusing. Maybe `[obj1, obj2, ... , objN]` would make it a bit more obvious what you're going for.